pandas/core/apply.py

```diff
@@ -307,23 +306,36 @@ def apply_standard(self):
                 # reached via numexpr; fall back to python implementation
                 pass
             else:
-                return self.obj._constructor_sliced(result, index=labels)
+                # this means that the reduction was successful
+                if partial_result is None:
+                    return self.obj._constructor_sliced(result, index=labels)
+                else:
+                    if isinstance(partial_result, Series):
+                        partial_result = DataFrame.infer_objects(partial_result)
 
         # compute the result using the series generator
-        results, res_index = self.apply_series_generator()
+        results, res_index = self.apply_series_generator(partial_result)
 
         # wrap results
         return self.wrap_results(results, res_index)
 
-    def apply_series_generator(self) -> Tuple[ResType, "Index"]:
+    def apply_series_generator(self, partial_result=None) -> Tuple[ResType, "Index"]:
         series_gen = self.series_generator
         res_index = self.result_index
 
         keys = []
```

Comment:
Am i missing something or is keys really unused?

Reply:
you can try to remove it (its possible its unused)